pkg/runcfanotify/runcfanotify.go

```diff
@@ -173,13 +165,28 @@ func NewRuncNotifier(callback RuncNotifyFunc) (*RuncNotifier, error) {
 	}
 	n.runcBinaryNotify = runcBinaryNotify
 
-	for _, file := range runcPaths {
-		err = runcBinaryNotify.Mark(unix.FAN_MARK_ADD, unix.FAN_OPEN_EXEC_PERM, unix.AT_FDCWD, file)
-		if err == nil {
-			log.Debugf("Checking %q: done", file)
-		} else {
-			log.Debugf("Checking %q: %s", file, err)
+	runcFound := false
+
+	for _, r := range runcPaths {
+		runcPath := filepath.Join(hostRoot, r)
+
+		log.Debugf("Runcfanotify: trying runc at %s", runcPath)
+
+		if _, err := os.Stat(runcPath); errors.Is(err, os.ErrNotExist) {
+			log.Debugf("Runcfanotify: runc at %s not found", runcPath)
+			continue
 		}
+		runcFound = true
+
+		if err := runcBinaryNotify.Mark(unix.FAN_MARK_ADD, unix.FAN_OPEN_EXEC_PERM, unix.AT_FDCWD, runcPath); err != nil {
+			runcBinaryNotify.File.Close()
+			return nil, fmt.Errorf("fanotify mark: %w", err)
+		}
```

Comment:
I suppose you are returning here because there is no way this operation fails on one path but succeeds on another, right?